x-pack/plugins/session_view/public/components/SessionView/index.tsx

```diff
@@ -46,18 +67,139 @@ export const SessionView = ({ sessionId }: SessionViewDeps) => {
     }
   };
 
+  const {
+    mutate
+  } = useMutation((insertData) => {
+    return http.put(INTERNAL_TEST_ROUTE, { 
+      body: JSON.stringify({ 
+        index: 'process_tree',
+        data: JSON.stringify(insertData)
+      })
+    });
+  });
+
+  const {
+    data: getData
+  } = useQuery(['process-tree', 'process_tree'], () =>
+    http.get(INTERNAL_TEST_ROUTE, {
+      query: {
+        index: 'process_tree',
+      },
+    })
+  );
+
+  const {
+    mutate: deleteMutate
+  } = useMutation((insertData) => {
+    return http.delete(INTERNAL_TEST_ROUTE, { 
+      body: JSON.stringify({ 
+        index: 'process_tree'
+      })
+    });
+  });
+  
+  const handleMutate = (insertData: any) => {
+    mutate(insertData, {
+      onSuccess: () => {
+        setData([...data, ...insertData])
+      }
+    });
+  }
+
+  const handleDelete = () => {
+    deleteMutate(undefined, {
+      onSuccess: () => {
+        setData([]);
+      }
+    })
+  }
+
+  useEffect(() => {
+    if (!getData) {
+      return;
+    }
+
+    if (getData.length <= data.length) {
+      return;
+    }
+
+    setData(getData.hits.map((event: any) => event._source));
+  }, [getData]);
+
+  const renderNoData = () => {
+    if (data.length) {
+      return null;
+    }
+    return (
+      <EuiEmptyPrompt 
+        title={<h2>No data to render</h2>}
+        body={
+          <p>
+            Please start by adding some data using the bottons below
+          </p>
+        }
+      />
+    )
+  }
+
+  const renderInsertButtons = () => {
+    return (
+      <EuiFlexGroup justifyContent="spaceAround">
+        <EuiFlexItem>
+          <EuiButton onClick={() => handleMutate(getStart())}>
+            Insert Session Start
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiButton onClick={() => handleMutate(getEvent())}>
+            Insert Command
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiButton onClick={() => handleMutate(getEnd())}>
+            Insert End
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiButton onClick={handleDelete}>
+            Delete Data
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    )
+  }
+
   return (
-    <div>
-      <EuiSearchBar query={searchQuery} onChange={onSearch} />
-      <div css={processTreeCSS}>
-        <ProcessTree
-          sessionId={sessionId}
-          forward={mockData}
-          searchQuery={searchQuery}
-          selectedProcess={selectedProcess}
-          onProcessSelected={onProcessSelected}
+    <EuiPage>
+      <EuiPageContent>
+        <EuiPageHeader 
```

Comment:
should this piece live in SessionViewPage?

Reply:
Yea it definitely could, although I envisioned the `SessionViewPage` to be a top level "route" page that just scaffolds whatever the main component is for that route, but yea we don't have any consensus there in terms of how to structure these things

Reply:
ya all good. I'm sure this stuff will change as we go.